typing/typedecl.ml

```diff
@@ -1401,22 +1413,30 @@ let transl_with_constraint env id row_path orig_decl sdecl =
         let cty = transl_simple_type env no_row sty in
         Some cty, Some cty.ctyp_type
   in
+  (* In the second part, we check the consistency between the two
+     declarations and compute a "merged" declaration; we now need to
+     work in the larger signature environment [sig_env], as that
+     [sig_decl.type_params] and [sig_decl.type_kind] are only valid
+     there. *)
+  let env = sig_env in
+  if arity_ok then
+    List.iter2 (Ctype.unify_var env) params sig_decl.type_params;
```

Comment:
I don't understand what you mean. Was it not possible to get a unification error before? Should we do something specific about unification errors, and if so, why?

Reply:
This is a question of unification order: since the `ptype_params` have to be distinct type variables, you cannot get a unification error by unifying them with constrained parameters. However, this could cause an error when unifying the `ptype_cstrs`, which would be reported as an `Inconsistent_constraint` error. So you need to wrap this call to `unify` in the same way.